### PR TITLE
Fix hco system health metric values

### PR DESCRIPTION
**Pull request by avlitman**
**What this PR does / why we need it**:
kubevirt_hco_system_health_status values changed to value+1,
this pr: 

- remove unknown from the values options
- metric values is changeed back the values to be 0 for healthy, 1 for warning and 2 for error. 
- remove reason label
- change alert name from OperatorConditionsUnhealth to HCOOperatorConditionsUnhealth.

also tests and related recording rules and alerts updated accordingly.

updated metric (before was 1):
Screenshot from 2024-12-23 20-06-01

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
jira-ticket: https://issues.redhat.com/browse/CNV-52735

```

**Release note**:

```release-note
none

```

### CODE REVIEW

File: pkg/monitoring/metrics/operator_metrics.go [hunk 1]
```diff
@@ -119,30 +117,20 @@ func IsHCOMetricHyperConvergedExists() (bool, error) {
...
-}
-
-func SetHCOSystemError(reason string) {
-	systemHealthStatus.Reset()
-	systemHealthStatus.WithLabelValues(reason).Set(SystemHealthStatusError)
+func SetHCOMetricSystemHealthStatus(status float64) {
+	systemHealthStatus.Set(status)
 }
 
 // GetHCOMetricSystemHealthStatus returns current value of gauge. If error is not nil then value is undefined
```
Comment (sradco) [inline pkg/monitoring/metrics/operator_metrics.go:124]: @avlitman is the comment here correct?

File: pkg/monitoring/rules/alerts/health_alerts.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ func healthAlerts() []promv1.Rule {
 			Alert: "OperatorConditionsUnhealthy",
 			Expr:  intstr.FromString(fmt.Sprintf("kubevirt_hco_system_health_status == %f", metrics.SystemHealthStatusError)),
 			Annotations: map[string]string{
-				"description": "HCO and its secondary resources are in a critical state due to {{ $labels.reason }}.",
+				"description": "HCO and its secondary resources are in a critical state due to system error.",
```
Comment (sradco) [inline pkg/monitoring/rules/alerts/health_alerts.go:18]: You can add here and below:
Go to Installed Operators -> kubevirt-hyperconverged-operator -> HyperConverged details and inspect the Conditions.

File: pkg/monitoring/metrics/operator_metrics.go (hunk 1 above)
Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics.go:124]: Yes it was before the changes to the metric

File: pkg/monitoring/rules/alerts/health_alerts.go (hunk 2 above)
Comment (avlitman) [reply pkg/monitoring/rules/alerts/health_alerts.go]: +1 added

File: pkg/monitoring/metrics/operator_metrics.go (hunk 1 above)
Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics.go:124]: which maybe problematic now that I think about it to return 0.. but this is how it was before any changes to the metric. WDYT?

Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics.go:124]: Before the code refactor it look like this:

```
// GetHCOMetricSystemHealthStatus returns current value of gauge. If error is not nil then value is undefined
func (hm *hcoMetrics) GetHCOMetricSystemHealthStatus() (float64, error) {
	return hm.GetMetricValue(HCOMetricSystemHealthStatus, nil)
}
```

so we can just drop the if maybe

File: pkg/monitoring/metrics/operator_metrics_test.go [hunk 3]
```diff
@@ -9,20 +9,16 @@ import (
 
 var _ = Describe("Operator Metrics", func() {
 	Context("kubevirt_hco_system_health_status", func() {
-		It("should set system error reason correctly", func() {
-			metrics.SetHCOSystemError("Reason1")
-			v, err := metrics.GetHCOMetricSystemHealthStatus("Reason1")
+		It("should set the correct system health status", func() {
+			metrics.SetHCOMetricSystemHealthStatus(2)
```
Comment (nunnatsa) [inline pkg/monitoring/metrics/operator_metrics_test.go:13]: Please use the constant: `metrics.SystemHealthStatusError`

File: pkg/monitoring/metrics/operator_metrics_test.go [hunk 4]
```diff
@@ -9,20 +9,16 @@ import (
...
+			v, err := metrics.GetHCOMetricSystemHealthStatus()
 			Expect(err).ToNot(HaveOccurred())
 			Expect(v).To(Equal(metrics.SystemHealthStatusError))
 
-			metrics.SetHCOSystemError("Reason2")
-			v, err = metrics.GetHCOMetricSystemHealthStatus("Reason2")
+			metrics.SetHCOMetricSystemHealthStatus(1)
```
Comment (nunnatsa) [inline pkg/monitoring/metrics/operator_metrics_test.go:18]: Please use the constant: `metrics.SystemHealthStatusWarning`

**Review by nunnatsa — COMMENTED** [review]
nit comments, inline

File: pkg/monitoring/metrics/operator_metrics.go (hunk 1 above)
Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics.go:124]: anyway this function is called only for testing purposes

File: pkg/monitoring/metrics/operator_metrics_test.go (hunk 3 above)
Comment (nunnatsa) [reply pkg/monitoring/metrics/operator_metrics_test.go:13]: it's the same thing. the constant is just a name for the value. but with the constant, the code is more readable.

Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics_test.go:13]: I mean what it will give me here:
 Expect(v).To(Equal(metrics.SystemHealthStatusError))

Comment (nunnatsa) [reply pkg/monitoring/metrics/operator_metrics_test.go:13]: The same. `metrics.SystemHealthStatusError` is 2

Comment (nunnatsa) [reply pkg/monitoring/metrics/operator_metrics_test.go:13]:
```go
const (
	SystemHealthStatusHealthy float64 = iota
	SystemHealthStatusWarning
	SystemHealthStatusError
)
```
See here: https://go.dev/ref/spec#Iota

Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics_test.go]: sure, fixed thanks!

File: pkg/monitoring/metrics/operator_metrics_test.go (hunk 4 above)
Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics_test.go]: fixed

**Comment by sradco** [comment]
@avlitman Please update the description and mention that the alert name was change.

**Comment by sradco** [comment]
@avlitman please review https://github.com/kubevirt/monitoring/pull/277 since without it merged, this PR test will fail.

**Comment by nunnatsa** [comment]
//rebase

**Pull request merged**